lib/ansible/playbook/role/__init__.py

```diff
@@ -222,6 +222,10 @@ def _load_role_data(self, role_include, parent_role=None):
                 raise AnsibleParserError("The handlers/main.yml file for role '%s' must contain a list of tasks" % self._role_name,
                                          obj=handler_data, orig_exc=e)
 
+        directories = [directory for directory in self._loader.list_directory(self._role_path) if directory not in ('tasks', 'handlers', 'meta')]
```

Comment:
For dependency-grouping there will be `metadata`, right?

The list of directories are those, that are already parsed by the role init: meta, tasks, handlers. All other directories are assumed to have non-trivial content and thus the role considered non-empty when they exist.